### PR TITLE
Port to Material UI V1 Beta36

### PR DESCRIPTION
**Pull request by Sunderous**
Greetings. This is my first time submitting a PR, so if I botched anything just let me know and I'll clean it up.

I was thinking of using this with a project, and wanted to use the new material UI, so I took a stab at updating it.

Browser still throws some warning from the material-ui-dots package, but looks like it functions properly in desktop, mobile, and mobile-landscape modes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,19 +33,22 @@
     "babel-cli": "^6.14.0",
     "babel-core": "^6.8.0",
     "babel-eslint": "^7.2.3",
+    "babel-plugin-direct-import": "^0.5.0",
     "babel-plugin-material-ui": "^0.3.0",
```
Comment (leMaik) [inline package.json]: babel-plugin-material-ui is not used anymore and can be removed

**Comment by Sunderous** [comment]
Took a quick look at this yesterday/today. I'm not sure I'll have time in the near future to do this properly (and styling is definitely my weakest area in front-end haha).

It looks like consolidating the inline styles into a stylesheet/classname list that gets passed through withStyles() might be a little tough, especially the computed styles. Could probably make a bunch of separate style names (desktop-root, mobile-root, etc.), and select based on desktop/mobile/landscape.

Might be worth waiting a little bit, after reading: https://github.com/mui-org/material-ui/issues/8726

It sounds like they're looking into updating how withStyles() works to handle react-jss styles that are dynamic based on their props. Which would greatly simplify handling tweaking the styles based on desktop/mobile/landscape.

If I'm overthinking this, and you had a different approach in mind, or can give me an example of how you'd want to port something non-trivial like:

```javascript
style={{
          pointerEvents: this.props.open ? null : 'none',
          opacity: this.props.open ? '1' : '0',
          ...this.props.style
        }}
```

to a withStyles() approach, hit me with it and I can take another look. If I do have some time this week, I might try and do it with the existing withStyles just to see what I come up with.

**Comment by leMaik** [comment]
@Sunderous The issue you linked is half a year old and closed. :thinking: Don't worry though, I will update the styles if you don't have the time. Your PR is a great start, I'll merge it and start building upon it. :+1:

**Comment by leMaik** [comment]
GitHub doesn't show it, but I merged this PR manually into the `next` branch.

**Comment by jaulz** [comment]
Any chance to push it to NPM? :)

**Comment by leMaik** [comment]
@jaulz `npm i material-auto-rotating-carousel@next` :wink:

**Comment by leMaik** [comment]
@jaulz The port is not finished, yet. :confused: I'll continue as soon as I have time.

**Comment by Asher978** [comment]
any updates? would love to use this.

**Comment by leMaik** [comment]
@Asher978 Not yet done, but it's about 80% there. :+1:

**Comment by leMaik** [comment]
@jaulz @Asher978 Done! :tada: `npm i --save material-auto-rotating-carousel@next`. Docs are here on the `next` branch, let me know what you think. :+1:

**Pull request closed without merging**